cobbler/modules/serializers/file.py

```diff
@@ -128,7 +145,7 @@ def deserialize_raw(collection_types):
         return results
 
 
-def filter_upgrade_duplicates(file_list):
+def filter_upgrade_duplicates(file_list) -> list:
```

Comment:
Could you please check if that function can be safely removed? Codacy complains that it is unused and a quick grep did not show any usages. Would be great to get rid of it OR use it at the appropriate places.

Reply:
It was introduced with #1245 in 2014 but is nowhere used [afaik](https://github.com/cobbler/cobbler/search?q=filter_upgrade_duplicates). Codacy complains about `what()`, too. There are many declarations of the `what()` function throughout Cobbler so I am not sure if Codacy is right here, too.

Reply:
The `what()` part should stay if my memory doesn't betray me because that should be used dynamically which these static checkers don't recognize.

I would vote for removing `filter_upgrade_duplicates()` then.

Reply:
Makes sense. Should we wait for feedback from the others or can I remove the function?

Reply:
What are your thoughts on this @brejoc, @hbokh?

Reply:
I'd vote vor removing. Our CI should catch it and if not then we can easily restore it from git.